src/module_1/module_1_meteo_api.py

```diff
@@ -1,7 +1,133 @@
-""" This is a dummy example """
+import requests
+import pandas as pd
+import time
+import plotly.graph_objects as go
+from plotly.subplots import make_subplots
+
+API_URL = "https://archive-api.open-meteo.com/v1/archive?"
+COORDINATES = {
+    "Madrid": {"latitude": 40.416775, "longitude": -3.703790},
+    "London": {"latitude": 51.507351, "longitude": -0.127758},
+    "Rio": {"latitude": -22.906847, "longitude": -43.172896},
+    }
+VARIABLES = ["temperature_2m_mean", "precipitation_sum", "wind_speed_10m_max"]
+
+def call_api(url:str, cool_off_time:int = 1):
+    """
+    Generic function to call API's and treats possible errors.
```

Comment:
Too generic docstring. It doesn't handle possible errors beyond retrying on rate limite. And that's fine, but if that's what we do, let's be more specific here. It's not a generic function either since it only does GET operations. 

Btw, I don't think this docstring adds any valuable information if the function had a good name `get_api_with_retries`